### DISCUSSION TITLE
Pydantic 2.0 breaks from_pydantic method of Configuration provider

### DISCUSSION THREAD

**Discussion by nybarius** [ets-labs/python-dependency-injector#730] [Q&A]
when using Pydantic-Settings. The fix seems pretty simple but bump-pydantic (https://github.com/pydantic/bump-pydantic) does not work due to the Py-C (I think).

**Comment by benm5678** [comment]
I'm facing the same.  Is there any alternative to this python dependency injector package?  It no longer seems to be maintained, yet still comes up as the top choice when searching.  Is DI not common practice in python?

**Reply by bmcandr** [reply]
FWIW, a DI framework like this is not required for implementing the DI pattern. Python supports DI out-of-the-box, but this framework helps take the legwork out of it and enables more "advanced" use cases like configuration management, mocking for tests, etc.

----

I'm also disappointed that maintenance has stalled on this package, but the core maintainer(s) appear to be Ukrainian so they may be focusing on more existentially important work at this time. Slava Ukraini!
